topology-manager/src/main/java/org/apache/aries/rsa/topologymanager/importer/TopologyManagerImport.java

```diff
@@ -105,105 +96,86 @@ public void remove(RemoteServiceAdmin rsa) {
     @Override
     public void remoteAdminEvent(RemoteServiceAdminEvent event) {
         if (event.getType() == RemoteServiceAdminEvent.IMPORT_UNREGISTRATION) {
-            unImport(event.getImportReference());
+            unimportService(event.getImportReference());
         }
     }
 
-    private void triggerSynchronizeImports(final String filter) {
+    private void synchronizeImportsAsync(final String filter) {
         LOG.debug("Import of a service for filter {} was queued", filter);
         if (!rsaSet.isEmpty()) {
-            execService.execute(new Runnable() {
-                public void run() {
-                    synchronizeImports(filter);
-                }
-            });
+            execService.execute(() -> synchronizeImports(filter));
         }
     }
-    
+
+    /**
+     * Synchronizes the actual imports with the possible imports for the given filter,
+     * i.e. unimports previously imported endpoints that are no longer possible,
+     * and imports new possible endpoints that are not already imported.
+     *
+     * @param filter the filter whose endpoints are synchronized
+     */
     private void synchronizeImports(final String filter) {
         try {
-            unImportForGoneEndpoints(filter);
-            importServices(filter);
+            Set<EndpointDescription> possible = importPossibilities.get(filter);
```

Comment:
I like the move to streams but I think I would keep the methods unImportForGoneEndpoints(filter) and       importServices(filter) they nicely split up the method.

Reply:
Can they be called importAddedEndpoints and unimportRemovedEndpoints or some better name?

Reply:
Sure. I did not want to mandate the naming. Just the idea to separate into two methods.